addons/l10n_vn_edi/views/account_move_views.xml

```diff
@@ -0,0 +1,84 @@
+                    <group>
+                        <field name="l10n_vn_edi_replacement_origin" invisible="1"/>
+                        <group>
+                            <field name="l10n_vn_edi_invoice_symbol" readonly="l10n_vn_edi_invoice_number"/>
+                            <field name="l10n_vn_edi_issue_date" invisible="not l10n_vn_edi_invoice_number"/>
+                        </group>
+                        <group>
+                            <field name="l10n_vn_edi_state"/>
+                            <field name="l10n_vn_edi_invoice_number" invisible="not l10n_vn_edi_invoice_number"/>
+                            <field name="l10n_vn_edi_reservation_code" invisible="not l10n_vn_edi_invoice_number"/>
+                        </group>
+                        <group string="Agreement"
+                               invisible="(not l10n_vn_edi_replacement_origin or not l10n_vn_edi_replacement_origin.l10n_vn_edi_invoice_number)
+                               and (not reversed_entry_id or not reversed_entry_id.l10n_vn_edi_invoice_number)">
+                            <field name="l10n_vn_edi_adjustment_type" readonly="l10n_vn_edi_invoice_number"/>
+                            <field name="l10n_vn_edi_agreement_document_name" readonly="l10n_vn_edi_invoice_number"/>
+                            <field name="l10n_vn_edi_agreement_document_date" readonly="l10n_vn_edi_invoice_number"/>
+                        </group>
+                    </group>
+                </page>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_account_invoice_filter_inherit_l10n_in_edi" model="ir.ui.view">
+        <field name="name">account.invoice.select.inherit.l10n.in.edi</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_account_invoice_filter"/>
+        <field name="arch" type="xml">
+            <filter name="late" position="after">
+                <separator/>
+                <field name="l10n_vn_edi_state"/>
+                <filter string="Need payment state update" name="pay_state_to_update" domain="[('l10n_vn_edi_state', '=', 'payment_state_to_update')]"/>
+            </filter>
+        </field>
+    </record>
+
+    <record id="view_invoice_tree_inherit_l10n_in_edi" model="ir.ui.view">
+        <field name="name">account.invoice.tree.inherit.l10n.in.edi</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_invoice_tree"/>
+        <field name="arch" type="xml">
+            <field name="state" position="after">
+                <field name="l10n_vn_edi_state"
+                       widget="badge"
+                       decoration-success="l10n_vn_edi_state == 'sent'"
+                       decoration-warning="l10n_vn_edi_state == 'ready_to_send'"
+                       decoration-info="l10n_vn_edi_state == 'payment_state_to_update'"
+                       optional="show"/>
```

Comment:
I've looked and can't find a way to do it, but would there not be a way to hide this column if the company in not a VN company ? It is not relevant to display it for other installed companies if they belong to other countries

Reply:
There is already the `optional="show"` that kind of does that.
Then we can have a `invisible` based on `country_code`, this should be an easy quick win.
And finally (but we've never done that AFAIK) we could have a field `env_country_codes` on `account.move.line`, and do a `column_invisible` based on that.
There is also a task somewhere for the framework to have features (a bit like security groups), and the country would be a feature and activate the column based on the environment.

Reply:
I didn't add the invisible here because to me it's not very useful. (the column shows anyway)
If the country of the invoice is not VN it will most likely have no value in this field (so it'll show empty in the col just like with invisible) and if there is a value for any non-standard reason we could want to display it.

The column_invisible could be a nice addition though, if we want to do that.